### PR TITLE
Faster ObjectGraph => JSON

### PR DESCRIPTION
**Pull request by mdittmer**
Run main ObjectGraph => JSON transformations in parallel. The new limiting factor is indexing records output by the transformation so they can be sorted before being stored.

### CODE REVIEW

File: lib/web_catalog/object_graph_importer.es6.js [hunk 1]
```diff
@@ -45,38 +54,116 @@ foam.CLASS({
...
+        });
+      },
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.box.Context',
+      name: 'apiContext_',
+      factory: function() {
+        let ctx = this.Context.create({
+        unsafe: false,
```
Comment (arobins) [inline lib/web_catalog/object_graph_importer.es6.js]: +2 indent here and next two lines

File: lib/web_catalog/object_graph_importer.es6.js [hunk 2]
```diff
@@ -45,38 +54,116 @@ foam.CLASS({
...
+        const numWorkers = this.numWorkers;
+        let workers = new Array(numWorkers);
+
+        for (let i = 0; i < numWorkers; i++) {
+          // Stub BoxRegistry interface over box addressed to new fork.
+          workers[i] = this.stubFactory.get(this.BoxRegistry).create({
+            delegate: this.ForkBox.create({
+              critical: true,
+              nodeParams:
+              // ['--inspect', '--debug-brk'],
```
Comment (arobins) [inline lib/web_catalog/object_graph_importer.es6.js]: Don't think you need the linebreak here

File: lib/web_catalog/object_graph_importer.es6.js [hunk 3]
```diff
@@ -45,38 +54,116 @@ foam.CLASS({
...
   ],
+
   methods: [
     {
       name: 'import',
       documentation: `Reads object graph files from objectGraphPath
           and extract web interfaces and import it to cloudstoreDAO
           using apiImporter.`,
       code: function() {
+        const ogFileRegExp = /window_[a-zA-Z0-9_.]+_[a-zA-Z0-9_.]+_[a-zA-Z0-9_.]+_[a-zA-Z0-9_.]+[.]json$/;
```
Comment (arobins) [inline lib/web_catalog/object_graph_importer.es6.js]: This makes me think that the collection of filenames could be its own method, or maybe an input to this object. The filename schema has to be right for the parsing to work, which is a bit ugly, but I think that can be addressed through documentation.

File: main/og_to_json.es6.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
```
Comment (arobins) [inline main/og_to_json.es6.js]: Document what this script does.

File: lib/web_catalog/object_graph_importer.es6.js (hunk 1 above)
Comment (mdittmer) [reply lib/web_catalog/object_graph_importer.es6.js]: Done (next push).

File: lib/web_catalog/object_graph_importer.es6.js (hunk 2 above)
Comment (mdittmer) [reply lib/web_catalog/object_graph_importer.es6.js]: This was for debugging. Done (next push).

File: lib/web_catalog/object_graph_importer.es6.js (hunk 3 above)
Comment (mdittmer) [reply lib/web_catalog/object_graph_importer.es6.js]: I've done my best to document this where `objectGraphPath` is defined. I got halfway through refactoring this into accepting an array of paths before realizing that the file name format is being used to extract information, too. I think if this importer gets refactored in the future, that would be a good time to consider other ways to extract the information and make this interface more robust.

File: main/og_to_json.es6.js (hunk 4 above)
Comment (mdittmer) [reply main/og_to_json.es6.js]: Done.

**Comment by mdittmer** [comment]
Comments addressed (and/or responded inline). @arobins PTAL.

**Pull request merged**